src/main/minio.js

```diff
@@ -2036,6 +2038,63 @@ export class Client {
         .on('end', () => cb(null, bucketNotification))
     })
   }
+
+  // Listens for bucket notifications. Returns an EventEmitter.
+  listenBucketNotification(bucketName, notificationARN) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError(`Invalid bucket name: ${bucketName}`)
+    }
+    if (typeof notificationARN !== 'string') {
+      throw new TypeError('notificationARN must be of type string')
+    }
+
+    let emitter = new EventEmitter()
+
+    // Listening constitutes repeatedly requesting s3 whether or not any
+    // changes have occurred. If we receive an 'end' event, stop.
+
+    let end = false
+    emitter.on('end', () => { end = true })
```

Comment:
nodejs devs will expect to do `emitter.removeListener()` to stop notifications so you will have to do it the nodejs idiomatic way